scripts/modules/ai-services-unified.js

```diff
@@ -716,16 +732,70 @@ async function generateTextService(params) {
  * @param {string} [params.systemPrompt] - Optional system prompt.
  * @param {string} params.commandName - Name of the command invoking the service.
  * @param {string} [params.outputType='cli'] - 'cli' or 'mcp'.
- * @returns {Promise<object>} Result object containing the stream and usage data.
+ * @returns {Promise<object>} Result object containing the stream and telemetry data.
  */
 async function streamTextService(params) {
 	const defaults = { outputType: 'cli' };
 	const combinedParams = { ...defaults, ...params };
-	// TODO: Validate commandName exists?
-	// NOTE: Telemetry for streaming might be tricky as usage data often comes at the end.
-	// The current implementation logs *after* the stream is returned.
-	// We might need to adjust how usage is captured/logged for streams.
-	return _unifiedServiceRunner('streamText', combinedParams);
+
+	const result = await _unifiedServiceRunner('streamText', combinedParams);
+
+	// Set up async telemetry capture for streaming (doesn't block return)
+	if (
+		result.mainResult &&
+		result.mainResult.usage &&
+		typeof result.mainResult.usage.then === 'function'
+	) {
+		// The usage property is a promise that resolves when the stream completes
+		result.mainResult.usage
+			.then((usageData) => {
+				// Update telemetry data with real usage information
+				if (usageData && result.telemetryData) {
+					result.telemetryData.inputTokens =
+						usageData.promptTokens || result.telemetryData.inputTokens || 0;
```

Comment:
for this whole chunk, wondering if its worth integrating inside the baseAI provider so that it can use the variables in there and also can be re-used more easily.

The idea of ai-services-unified is to unify the ai services, adding extra logic in here doesn't make sense.

So maybe we move the whole streamTextService ?

Reply:
Not sure, though not a blocker to merge this, so we can figure it out in a separate PR